internal/testrunner/runners/system/servicedeployer/kubernetes.go

```diff
@@ -0,0 +1,269 @@
+	fileInfos, err := ioutil.ReadDir(definitionsDir)
+	if err != nil {
+		return nil, errors.Wrapf(err, "can't read definitions directory (path: %s)", definitionsDir)
+	}
+
+	var definitionPaths []string
+	for _, fileInfo := range fileInfos {
+		if strings.HasSuffix(fileInfo.Name(), ".yaml") {
+			definitionPaths = append(definitionPaths, filepath.Join(definitionsDir, fileInfo.Name()))
+		}
+	}
+	return definitionPaths, nil
+}
+
+func verifyKindContext() error {
+	logger.Debug("ensure that kind context is selected")
+
+	cmd := exec.Command("kubectl", "config", "current-context")
+	errOutput := new(bytes.Buffer)
+	cmd.Stderr = errOutput
+	output, err := cmd.Output()
+	if err != nil {
+		return errors.Wrapf(err, "kubectl command failed (stderr=%q)", errOutput.String())
+	}
+	currentContext := string(bytes.TrimSpace(output))
+
+	if currentContext != "kind-kind" {
+		return fmt.Errorf("unexpected kubectl context selected (actual: %s, expected: %s)", currentContext, kindContext)
+	}
+	return nil
+}
+
+func findKindControlPlane() (string, error) {
+	logger.Debugf("find \"%s\" container", kindControlPlaneContainerName)
+
+	cmd := exec.Command("docker", "ps", "--filter", "name=kind-control-plane", "--format", "{{.ID}}")
+	errOutput := new(bytes.Buffer)
+	cmd.Stderr = errOutput
+	output, err := cmd.Output()
+	if err != nil {
+		return "", errors.Wrapf(err, "could not find \"%s\" container to the stack network (stderr=%q)", kindControlPlaneContainerName, errOutput.String())
+	}
+	containerIDs := bytes.Split(bytes.TrimSpace(output), []byte{'\n'})
+	if len(containerIDs) != 1 {
+		return "", fmt.Errorf("expected single %s container, make sure you have run \"kind create cluster\" and the %s container is present", kindControlPlaneContainerName, kindControlPlaneContainerName)
+	}
+	return string(containerIDs[0]), nil
+}
+
+func connectControlPlaneToElasticStackNetwork(controlPlaneContainerID string) error {
+	stackNetwork := fmt.Sprintf("%s_default", stack.DockerComposeProjectName)
+
+	logger.Debugf("check network connectivity between service container %s (ID: %s) and the stack network %s", kindControlPlaneContainerName, controlPlaneContainerID, stackNetwork)
+	cmd := exec.Command("docker", "network", "inspect", stackNetwork)
+	errOutput := new(bytes.Buffer)
+	cmd.Stderr = errOutput
+	output, err := cmd.Output()
+	if err != nil {
+		return errors.Wrapf(err, "could not inspect the stack network (stderr=%q)", errOutput.String())
+	}
+
+	var networkDescriptions []struct {
+		Containers map[string]struct {
+			Name string
+		}
+	}
+	err = json.Unmarshal(output, &networkDescriptions)
+	if err != nil {
+		return errors.Wrapf(err, "can't unmarshal network inspect for %s (stderr=%q)", stackNetwork, errOutput.String())
+	}
+
+	if len(networkDescriptions) != 1 {
+		return fmt.Errorf("expect single network inspect record, got %d entries", len(networkDescriptions))
+	}
+
+	for _, c := range networkDescriptions[0].Containers {
+		if c.Name == kindControlPlaneContainerName {
```

Comment:
Under what circumstances would the kind control plane container be **already** attached to the stack network?

Reply:
As the Elastic Agent is loaded lazily, since execution of the second system test, the container is already connected.

Reply:
I'm not opposed to this but I'm wondering if it would be better (clean slate approach) to instead utilize the `TearDown` method to detach the network and remove any containers started up in `SetUp` method?

Reply:
My intention was to cut the total execution time, not to waste time on booting up the agent, but I can adjust to it.

Reply:
Ah okay, let's keep it this way. If we see any issues we can adjust to do more teardown and re-setup.